versions.go

```diff
@@ -1,26 +1,66 @@
 /*
 MongoDB Versions
 
-The MongoDBVersion type provides support for interacting with MongoDB
+The LegacyMongoDBVersion type provides support for interacting with MongoDB
 versions. This type makes it possible to validate MongoDB version
 numbers and ask common questions about MongoDB versions.
 */
 package bond
 
 import (
 	"fmt"
+	"github.com/pkg/errors"
 	"sort"
 	"strconv"
 	"strings"
 
 	"github.com/blang/semver"
 )
 
-// MongoDBVersion is a structure representing a version identifier for
-// MongoDB. Use the associated methods to ask questions about MongoDB
+const endOfLegacy = "4.5.0"
+
+// MongoDBVersion encapsulates information about a MongoDBVersion.
+// Use the associated methods to ask questions about MongoDB
 // versions. All parsing of versions happens during construction, and
-// individual method calls are very light-weight.
-type MongoDBVersion struct {
+// individual method calls are very light-weight. Note that
+// not all methods are applicable for all versions.
+type MongoDBVersion interface {
+	// String returns a string representation of the MongoDB version number.
+	String() string
+	// Parsed returns the parsed version object for the version
+	Parsed() semver.Version
+	// Series returns the release series for legacy versions.
+	Series() string
+	// IsReleaseCandidate returns true if the version is a release candidate.
+	IsReleaseCandidate() bool
+	// IsStableSeries returns true if the legacy version is a stable series.
+	IsStableSeries() bool
+	// IsDevelopmentSeries returns true if the legacy version is a development series.
+	IsDevelopmentSeries() bool
+	// StableReleaseSeries returns true if the legacy version is a stable release series.
+	StableReleaseSeries() string
+	// IsRelease returns true if the version is a release.
+	IsRelease() bool
+	// IsDevelopmentBuild returns true for non-release versions.
+	IsDevelopmentBuild() bool
+	// IsInitialStableReleaseCandidate returns true if the legacy version is a release
+	// candidate for the initial release of a stable series.
+	IsInitialStableReleaseCandidate() bool
+	// RcNumber returns the RC counter (or -1 if not a release candidate)
+	RcNumber() int
+
+	IsLessThan(version MongoDBVersion) bool
+	IsLessThanOrEqualTo(version MongoDBVersion) bool
+	IsGreaterThan(version MongoDBVersion) bool
+	IsGreaterThanOrEqualTo(version MongoDBVersion) bool
+	IsEqualTo(version MongoDBVersion) bool
+	IsNotEqualTo(version MongoDBVersion) bool
+
+}
+
+// LegacyMongoDBVersion is a structure representing a version identifier for legacy versions of
+// MongoDB, which implements the MongoDBVersion interface.
+type LegacyMongoDBVersion struct {
```

Comment:
Should this be exported?

Reply:
I think it should be exported so that it's obvious what interface methods are supported/not supported on legacy vs. new versions.

Reply:
Good point